gtest/src/manager/block_exec.rs

```diff
@@ -345,86 +369,83 @@ impl ExtManager {
 
         match response {
             Ok(Mocked::Reply(reply)) => {
-                let maybe_reply_message = if let Some(payload) = reply {
-                    let id = MessageId::generate_reply(message_id);
-                    let packet = ReplyPacket::new(payload.try_into().unwrap(), 0);
-                    Some(ReplyMessage::from_packet(id, packet))
-                } else {
-                    (!dispatch.is_reply() && dispatch.kind() != DispatchKind::Signal)
-                        .then_some(ReplyMessage::auto(message_id))
+                let kind = DispatchResultKind::Success;
+                let (generated_dispatches, reply_sent) = reply
+                    .map(|payload| {
+                        let reply_message = ReplyMessage::from_packet(
+                            MessageId::generate_reply(dispatch.id()),
+                            ReplyPacket::new(payload.try_into().expect("too big payload"), 0),
+                        );
+                        let dispatch = reply_message.into_dispatch(
+                            program_id,
+                            dispatch.source(),
+                            dispatch.id(),
+                        );
+
+                        (vec![(dispatch, 0, None)], true)
+                    })
+                    .unwrap_or_default();
+                let dispatch_result = DispatchResult {
+                    kind,
+                    dispatch,
+                    program_id,
+                    generated_dispatches,
+                    gas_amount: gas_counter.to_amount(),
+                    reply_sent,
+                    context_store: Default::default(),
+                    awakening: Default::default(),
+                    reply_deposits: Default::default(),
+                    program_candidates: Default::default(),
+                    gas_reserver: Default::default(),
+                    system_reservation_context: Default::default(),
+                    page_update: Default::default(),
+                    allocations: Default::default(),
```

Comment:
Maybe use `const` or function to default initialize these fields,
for ex.:
 
```rust
//.. 

 let dispatch_result = DispatchResult {
                    kind,
                    dispatch,
                    program_id,
                    generated_dispatches,
                    gas_amount: gas_counter.to_amount(),
                    reply_sent,
                   .. default_dispatch_result()
};

// ...

fn default_dispatch_result() -> DispatchResult {
    DispatchResult {
        kind: DispatchResultKind::Success,
        dispatch: Default::default(),
        program_id: Default::default(),
        gas_amount: GasCounter::new(0).to_amount(),
        context_store: Default::default(),
        awakening: Default::default(),
        generated_dispatches: Default::default(),
        reply_deposits: Default::default(),
        program_candidates: Default::default(),
        gas_reserver: Default::default(),
        system_reservation_context: Default::default(),
        page_update: Default::default(),
        allocations: Default::default(),
        reply_sent: Default::default(),
    }
}

```

And same below.

Reply:
done